packages/collector/test/tracing/cloud/aws-sdk/v2/sqs/receiveMessage.js

```diff
@@ -99,49 +99,54 @@ async function receiveAsync() {
   try {
     const sqsPromise = sqs.receiveMessage(receiveParams).promise();
     const data = await sqsPromise;
-    instana.sdk.runInAsyncContext(sqsPromise.instanaAsyncContext, async () => {
-      if (data && data.error) {
-        log('receive message data error', data.error);
-        return;
-      } else if (!data || !data.Messages || data.Messages.length === 0) {
-        log('no messages, doing nothing');
-        return;
-      }
 
-      data.Messages.forEach(message => {
-        sendToParent(message);
-      });
+    return new Promise(resolve => {
```

Comment:
This change makes sure that a promise is returned and "awaited" before the loop triggers another queue polling.

Reply:
This problem is also present in the current CLS. But as we manually control the active context in the `exit()` method, the [this._set.pop()](https://github.com/instana/nodejs/blob/main/packages/core/src/tracing/clsHooked/context.js#L294) call removes the active context (which contains the SQS exit span, if I remember correctly now).
This `.pop()` sets the active context to null (in the particular case of this test, there is nothing else in the `this._set` array), so the SQS instrumentation doesn't complain about an existent `currentSpan()` when it attempts to instrument the entry span.